File: frontend-lsport/src/app/entities/Jogador.ts
import {FootballElement} from "./FootballElement";

export class Jogador extends FootballElement {
  posicao: string;
  equipaNacional: string;
  pePreferido: string;
  dataNascimento: string;
  altura: number;
  nacionalidade: string;
  trofeus: string;

  constructor(
    id: number,
    nome: string,
    equipa: string,
    descricao: string,
    photoPath: string,
    posicao: string,
    equipaNacional: string,
    pePreferido: string,
    dataNascimento: string,
    altura: number,
    nacionalidade: string,
    trofeus: string
  ) {
    super(id, nome, equipa, descricao, photoPath);
    this.posicao = posicao;
    this.equipaNacional = equipaNacional;
    this.pePreferido = pePreferido;
    this.dataNascimento = dataNascimento;
    this.altura = altura;
    this.nacionalidade = nacionalidade;
    this.trofeus = trofeus;
  }
}

File: backend-lfsport/src/main/java/com/ngola/backendlfsport/entity/Jogador.java
package com.ngola.backendlfsport.entity;

import jakarta.persistence.Entity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
public class Jogador extends FootballElement{
    private String posicao;
    private String equipaNacional;
    private String pePreferido;
    private String dataNascimento;
    private int altura;
    private String nacionalidade;
    private String trofeus;
}
